informer/collector.go

```diff
@@ -0,0 +1,54 @@
+package informer
+
+import (
+	"context"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+var (
+	description *prometheus.Desc = prometheus.NewDesc(
+		prometheus.BuildFQName("operatorkit", "informer", "deletion_timestamp"),
+		"DeletionTimestamp of watched objects.",
+		[]string{
+			"exported_name",
+			"exported_namespace",
+		},
+		nil,
+	)
+)
+
+// Describe is used to describe metrics which are exported to Prometheus.
+func (i *Informer) Describe(ch chan<- *prometheus.Desc) {
+	ch <- description
+}
+
+// Collect is called by the Prometheus registry when collecting metrics from
+// the informer.
+func (i *Informer) Collect(ch chan<- prometheus.Metric) {
+	eventChan := make(chan watch.Event)
+	ctx := context.Background()
+	go func() {
+		err := i.fillCache(ctx, eventChan)
+		if err != nil {
+			return
```

Comment:
Not happy about this either, should we pass a logger into the informer? Was it a concious decision to not have a logger in here?

Reply:
I don't see a problem in doing that.

Reply:
Having logging for failures in the collector would be cool

Reply:
+1 on logging.

Reply:
Added a logger and some logging similar to the one we have in the aws collector.